README.md

```diff
@@ -23,23 +23,29 @@
 [bors-img]: https://bors.tech/images/badge_small.svg
 [bors-url]: https://app.bors.tech/repositories/32073
 
-A `DispatchedTuple` is like a dictionary, except
+## What are dispatched tuples?
 
- - the keys are **instances of types**
- - they are backed by tuples, so they are GPU-friendly
- - multiple keys are allowed (except for `DispatchedTupleSet`)
- - each unique key returns a tuple of values given by that key (in order)
+`DispatchedTuple`s are like immutable dictionaries (so, they're more like `NamedTuple`s) except that the keys are **instances of types**. Also, because `DispatchedTuple`s are backed by tuples, they are GPU-friendly.
 
-All `AbstractDispatchedTuple`s take a `Tuple` of `Pair`s, where the `first` field of the `Pair` (the "key") is **an instance of the type you want to dispatch on**. The `second` field of the `Pair` is the quantity (the "value", which can be anything) returned by `dispatch(::AbstractDispatchedTuple, key)`, the one user-facing method exported by DispatchedTuples.jl.
-
-DispatchedTuples.jl has several user-facing types:
+There are two kinds of `DispatchedTuple`s with different behavior:
 
- - `DispatchedTuple` - a dispatched tuple (example below)
+```
+┌────────────────────────────────────┬───────────────────────────┬────────────────────┐
+│                       Return value │           DispatchedTuple │ DispatchedTupleSet │
```

Comment:
Hmmm, why not `DispatchedTuple` and `DispatchedSet`?

Reply:
Maybe I should have opened an issue about this, it's not really within the scope of this PR.

Reply:
I think I'm fine with that name